### PR TITLE
CASMHMS-5180: Add xnametypes and xnames packaged to hms-xname

### PR DESCRIPTION
**Pull request by rsjostrand-hpe**
## Summary and Scope
hms-xname is the new home for all things xname related, such as types, enumerations, structures and functions for interacting with xnames.

Github actions run unit tests.

### xnametypes package:
  * CASMHMS-5180: 
     * Moved HMSTypes and related functions to the new under the xnametypes package. 
     * Added functions to expose the contents of the `hmsCompRecognitionTable`
  * CASMHMS-4667: 
     * GetHMSCompParent now returns s0 for CDUs and Cabinets.
  * CASMHMS-4668: 
    * Added missing MgmtHLSwitchEnclosure type
    * Corrected the parent type for MgmtHLSwitch to be MgmtHLSwitchEnclosure
    * Corrected the parent type for RouterTORFpga to be RouterTOR

### xnames package:
The xnames (xname **s**tructure/**s**tringify) package allows xnames to be manipulated via Go Structures, instead of using `fmt.Sprintf` or regular expressions to manipulate the string representation of xnames. This package also contains facilities to easily convert to and from the string representation of a xname to xnames structures.

The files types.go and util.go are generated from the contents  the `hmsCompRecognitionTable` from the xnametypes package.

## Issues and Related PRs
_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._
* Partially resolves [CASMHMS-5180](https://jira-pro.its.hpecorp.net:8443/browse/CASMHMS-5180)
* Resolves [CASMHMS-4667](https://jira-pro.its.hpecorp.net:8443/browse/CASMHMS-4667)
* Resolves [CASMHMS-4668](https://jira-pro.its.hpecorp.net:8443/browse/CASMHMS-4668)
* Related PR: https://github.com/Cray-HPE/hms-base/pull/16

## Testing

_List the environments in which these changes were tested._

### Tested on:
  * Local system

### Test description:
All existing unit tests from copied over from hms-base, and newly added unit tests pass. 

For CASMHMS-4667, added unit tests that the correct parent is returned when using GetHMSCompParent on a CDU or cabinet.

## Risks and Mitigations
Low risk. Services will need to explicitly upgrade to this new package to get these changes.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable
- [x] Update go.mod after https://github.com/Cray-HPE/hms-base/pull/16 is merged and tagged
- [x] Update run unit test workflow after https://github.com/Cray-HPE/hms-base/pull/16 is merged

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+require github.com/Cray-HPE/hms-base/v2 v2.0.0-00010101000000-000000000000
+
+require (
+	github.com/fsnotify/fsnotify v1.4.9 // indirect
+	github.com/hashicorp/go-cleanhttp v0.5.1 // indirect
+	github.com/hashicorp/go-retryablehttp v0.5.4 // indirect
+	golang.org/x/sys v0.0.0-20191005200804-aed5e4c7ecf9 // indirect
+)
+
+replace github.com/Cray-HPE/hms-base/v2 => ../hms-base
```
Comment (rsjostrand-hpe) [inline go.mod]: This line needs to get removed after https://github.com/Cray-HPE/hms-base/pull/16 is merged and tagged. Then need to run:
```bash
go mod tidy
go mod vendor
```

File: .github/workflows/run_unit_test.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+name: Run Unit Tests
+on: [push, pull_request]
+jobs:
+  run_unit_test:
+    uses: Cray-HPE/hms-build-scripts/.github/workflows/run_unit_test.yaml@develop
```
Comment (rsjostrand-hpe) [inline .github/workflows/run_unit_test.yaml]: Switch to the master branch when https://github.com/Cray-HPE/hms-build-scripts/pull/4 is merged

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
+### Summary and Scope
```
Comment (shunr-hpe) [inline .github/pull_request_template.md]: Recently, for new projects, we used the template at:
https://github.com/Cray-HPE/csm-template/blob/main/.github/pull_request_template.md

I have no preference on which one is used.

**Review by shunr-hpe — APPROVED** [review]

File: .github/pull_request_template.md (hunk 3 above)
Comment (rsjostrand-hpe) [reply .github/pull_request_template.md]: Updated

**Review by mjendrysik-hpe — APPROVED** [review]

**Pull request merged**